### ISSUE TITLE
Passing -gdwarf-5 generates neither .apple_names nor .debug_names

### ISSUE THREAD

**Issue by JDevlieghere** [llvm/llvm-project#53532]
On Apple platforms, we generate `.apple_names`, `.apple_types`, `.apple_namespaces` and `.apple_objc` Apple accelerator tables for DWARF 4 and earlier. For DWARF 5, I would expect us to generate `.debug_names`, but instead we get no accelerator tables at all.

In the backend we are correctly determining that we should be emitting `.debug_names` instead of `.apple_names`. However, when we get to the point of emitting the section, [CU debug name table kind is not "default" are skipped ](https://github.com/apple/llvm-project/blob/5e2798f25ca9587a2343e9df2c487a341718934b/llvm/lib/CodeGen/AsmPrinter/AccelTable.cpp#L553). This property is controlled by the `-gpubnames` argument. Indeed, passing `-gdwarf-5 -gpubnames` does generate `.debug_names` section, on all platforms.

On Apple platforms, the `.debug_names` section should be emitted by default.

**Comment by JDevlieghere** [comment]
CC @adrian-prantl @labath @dwblaikie @clayborg

**Comment by dwblaikie** [comment]
yep - makes sense to enable them by default when tuning for lldb I guess - not sure whether that would be implemented in clang or llvm, I forget where some of those determinations are made (I'd somewhat advocate for made in clang, so users could opt-out somehow, like -gno-pubnames might still override the tuning default, for instance)
